src/services/Config.ts

```diff
@@ -1,12 +1,51 @@
 import * as vscode from 'vscode';
-import { CONFIG_CORE_SERVERS, CONFIG_KEY } from '../common';
+import {
+  CONFIG_CORE_SERVERS,
+  CONFIG_KEY,
+  ConnectionConfig,
+  ConnectionConfigStored,
+  DEFAULT_CONSOLE_TYPE,
+} from '../common';
+import { InvalidConsoleTypeError, Logger } from '../util';
+
+const logger = new Logger('Config');
 
 function getConfig() {
   return vscode.workspace.getConfiguration(CONFIG_KEY);
 }
 
-function getCoreServers(): string[] {
-  return getConfig().get(CONFIG_CORE_SERVERS, []);
+function getCoreServers(): ConnectionConfig[] {
+  const config = getConfig().get<ConnectionConfigStored[]>(
+    CONFIG_CORE_SERVERS,
+    []
+  );
+
+  // Expand each server config to a full `ConnectionConfig` object.
+  const expandedConfig = config.map(server =>
+    typeof server === 'string'
+      ? { consoleType: DEFAULT_CONSOLE_TYPE, url: server }
+      : {
+          consoleType: DEFAULT_CONSOLE_TYPE,
+          ...server,
+        }
+  );
+
+  logger.info('Core servers:', JSON.stringify(expandedConfig));
+
+  return expandedConfig.filter(server => {
+    try {
+      // Filter out any invalid server configs to avoid crashing the extension
+      // further upstream.
+      new URL(server.url);
+      if (!['python', 'groovy'].includes(server.consoleType)) {
```

Comment:
Could define these supported languages as a set, and check if `LANGUAGE_SET.has(server.consoleType)`. No big deal really.